src/coreclr/tools/aot/ILCompiler/ILCompilerRootCommand.cs

```diff
@@ -160,6 +160,8 @@ internal sealed class ILCompilerRootCommand : RootCommand
             new(new[] { "--singlemethodgenericarg" }, "Single method compilation: generic arguments to the method");
         public Option<string> MakeReproPath { get; } =
             new(new[] { "--make-repro-path" }, "Path where to place a repro package");
+        public Option<string[]> RootedAssembliesExports { get; } =
+            new(new[] { "--rootedassembliesexports" }, Array.Empty<string>, "Root exported methods from assemblies");
```

Comment:
```suggestion
            new(new[] { "--rootunmanagedentrypoints" }, Array.Empty<string>, "Generate unmanaged entrypoints for given assembly");
```
I think this is a better description. There is not a thing called "exported method". The closest thing are [exported types](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.metadata.exportedtype) that are something completely different.

Reply:
I agree. Should I also: 
1. Change the name of the ItemGroup:
From: `IlcRootedAssembliesExports` into: `IlcRootUnmanagedEntryPoint`?
2. Change the name of https://github.com/dotnet/runtime/blob/6ddb03f8e8c1876fa14f0289a42c128dd95e08a4/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportedMethodsRootProvider.cs#L15
into `UnmanagedEntryPointsRootProvider`

Reply:
I think so.

Looking at the command line option again, I am wondering whether `--generateunmanagedentrypoints` / `GenerateUnmanagedEntryPoints` would be even better. `Generate` is more self-explanatory than `Root`.

Reply:
My initial idea was to keep the `root` in the name, as using this feature could affect trimming (as it could roots a method/type essentially). 

On the other hand, `generate` sounds (at least to me) like the compiler would do some extra work to expose something not intended to be exposed.

Finally, I know it is a very long word, but I believe something like: `ExportUnmanagedEntryPointsFromReferencedAssemblies` would be good for the ItemGroup, while the compiler switch can be shorten.

Reply:
> My initial idea was to keep the root in the name, as using this feature could affect trimming (as it could root a method/type essentially).

The standard rooting keeps methods visible to reflection, without necessarily generating unmanaged entrypoint for them. This option does extra work by generating the named unmanaged entrypoint for the method (without exposing the method to reflection in any way). These two options are different - neither is subset of the other.

I do not see any negative connotations in `generate`. There are many things called `Generate...` in the SDK and many of them are on by default.

The problem that I see with `Generate` is that it is a verb. It would be a good name for a task or a bool property. It is not a great name for ItemGroup. ItemGroup names should be nouns. So a better name can be something like `UnmanagedEntryPointsAssembly`.

> ExportUnmanagedEntryPointsFromReferencedAssemblies

Yes, it is a mouthful.

Reply:
`generateunmanagedentrypoints` sounds good to me. Also throwing `exportunmanagedentrypoints` as a suggestion.

Reply:
New naming introduced.